### ISSUE TITLE
RFE: derive package name from spec filename

### ISSUE THREAD

**Issue by pmatilai** [rpm-software-management/rpm#1208]
In the ever-quest for reducing redundant boilerplate from spec files: the de-facto standard in rpm world is to name the spec after its Name: tag. Which  begs the question: what do we need the Name: tag for then? We could just grab the name from the filename, which eliminates a redundant line from the spec, and eliminates chicken-egg issues around %{name} definition.

**Comment by rpm-maint** [comment]
Panu Matilainen <notifications@github.com> writes:

> In the ever-quest for reducing redundant boilerplate from spec files: the de-facto standard in rpm world is to name the spec after its Name: tag. Which  begs the question: what do we need the Name: tag for then? We could just grab the name from the filename, which eliminates a redundant line from the spec, and eliminates chicken-egg issues around %{name} definition.
>

If you'd still be able to override the name of the package via a Name:
tag, why not?

**Comment by ignatenkobrain** [comment]
I think this is great idea, though having `%{source_name}` or `%{spec_name}` (just bikeshedding) would be useful so that you could define `Name` tag in the way like `%{source_name}%{compat_version}` so that you can build different packages out of one spec file by just appending different suffixes and not having to rename spec.

But even if this is not going to happen, I am fine with having Name to be defined from spec name.

**Comment by pmatilai** [comment]
Yeah, the introducing a concept of "source name" seems essential if we want to preserve ability to override Name from spec - which obviously does have its uses. This separation is largely missing from rpm and is at the heart of various problems and bad de-facto practises.

One of the chicken-egg problems is that people expect the dist-git style file layout, which is easy to establish by basing the top/source/spec directories around `${PWD}`, *but* I think many people would want and expect `rpm -i foo.src.rpm` to land with the stuff in ~/topdir/foo instead of current dir. Which introduces %{name} into %_topdir or so, which isn't workable. But spec file name and src.rpm %{name} are always immediately available.

**Comment by voxik** [comment]
How this would work with SCLs which prefixes the `%{name}`?

**Comment by voxik** [comment]
And keep the original name in `%{pkg_name}`

**Comment by ignatenkobrain** [comment]
@voxik well, if you define `Name` in spec, nothing will change for you. But this is very similar to what I was asking few comments above to be able to redefine Name while preserving access to the auto-guessed one, so that for example buildsystem could just do `-D 'name %{scl_prefix}-%{source_name}'`

**Comment by voxik** [comment]
Yes, I understand this is similar question. I just want to bring one practical example, which has standardized way to changing the package name. So if there was possibility to reuse something, so SCLs would be simpler, that would be good.

**Comment by pmatilai** [comment]
> -D 'name %{scl_prefix}-%{source_name}'

That's not going to work, the Name: tag will need to appear in the spec to override it.

But yes, with the concept of a "source package" and following good practises in the spec, SCLs would become a (relatively) trivial exercise.
